tests/test_config.py

```diff
@@ -0,0 +1,179 @@
+	with pytest.raises(ValueError,match="No parameter function found"):
+		lux.register_action("bars")
+
+def test_q5_invalid_function():
+	df = pd.read_csv("lux/data/car.csv")
+	with pytest.raises(ValueError,match="Value must be a callable"):
+		lux.register_action("bars", "not a Callable")
+
+def test_q6_invalid_validator():
+	df = pd.read_csv("lux/data/car.csv")
+	def random_categorical(ldf):
+		intent = [lux.Clause("?",data_type="nominal")]
+		vlist = VisList(intent,ldf)
+		for vis in vlist:
+			vis.score = 10
+		vlist = vlist.topK(15)
+		return {"action":"bars", "description": "Random list of Bar charts", "collection": vlist}
+	with pytest.raises(ValueError,match="Value must be a callable"):
+		lux.register_action("bars", random_categorical, "not a Callable")
+
+def test_q7_remove_action():
+	def random_categorical(ldf):
+		intent = [lux.Clause("?",data_type="nominal")]
+		vlist = VisList(intent,ldf)
+		for vis in vlist:
+			vis.score = 10
+		vlist = vlist.topK(15)
+		return {"action":"bars", "description": "Random list of Bar charts", "collection": vlist}
+	def contain_horsepower(df):
+		for clause in df.intent:
+			if clause.get_attr() == "Horsepower":
+				return True
+		return False
+	df = pd.read_csv("lux/data/car.csv")
+	lux.register_action("bars", random_categorical, contain_horsepower)
+	df.set_intent(["Acceleration", "Horsepower"])
+	df._repr_html_()
+	assert("bars" in df.recommendation)
+	assert(len(df.recommendation["bars"]) > 0)
+	lux.remove_action("bars")
+	df._repr_html_()
+	assert("bars" not in df.recommendation)
+
+def test_q8_remove_invalid_action():
+	df = pd.read_csv("lux/data/car.csv")
+	with pytest.raises(ValueError,match="Option 'bars' has not been registered"):
+		lux.remove_action("bars")
+
+def test_q9_remove_default_actions():
+	df = pd.read_csv("lux/data/car.csv")
+	df._repr_html_()
+
+	lux.remove_action("Distribution")
+	df._repr_html_()
+	assert("Distribution" not in df.recommendation)
+
+	lux.remove_action("Occurrence")
+	df._repr_html_()
+	assert("Occurrence" not in df.recommendation)
+
+	lux.remove_action("Temporal")
+	df._repr_html_()
+	assert("Temporal" not in df.recommendation)
+
+	lux.remove_action("Correlation")
+	df._repr_html_()
+	assert("Correlation" not in df.recommendation)
+
+	assert(len(df.recommendation) == 0)
+
+	def random_categorical(ldf):
+		intent = [lux.Clause("?",data_type="nominal")]
+		vlist = VisList(intent,ldf)
+		for vis in vlist:
+			vis.score = 10
+		vlist = vlist.topK(15)
+		return {"action":"bars", "description": "Random list of Bar charts", "collection": vlist}
+	def contain_horsepower(df):
+		for clause in df.intent:
+			if clause.get_attr() == "Horsepower":
+				return True
+		return False
+	df = pd.read_csv("lux/data/car.csv")
+	lux.register_action("bars", random_categorical, contain_horsepower)
+	df.set_intent(["Acceleration", "Horsepower"])
+	df._repr_html_()
+	assert("bars" in df.recommendation)
+	assert(len(df.recommendation["bars"]) > 0)
```

Comment:
This is a really comprehensive test suite that tests many of the edge cases of how the `register_action` and `remove_action` might be used!